packages/next/src/server/config-shared.ts

```diff
@@ -20,9 +20,11 @@ export type NextConfigComplete = Required<NextConfig> & {
   configFileName: string
 }
 
+export type I18NDomains = DomainLocale[]
```

Comment:
Probably no need for this type. You can just re-export `DomainLocale` and wherever `I18NDomains` is needed, you do `DomainLocale[]` instead.

Reply:
I prefer having this explicit type instead of re-exporting DomainLocale as then you end up with scattered `DomainLocale[]` which specifically refers to `nextConfig.i18n.domains`. Could have also used `NextConfigComplete['i18n']['domains]'` with `Omit<>` but personally prefer adding additional types over trying to coerce the type into only what you need.